netlify/edge-functions/tests/serve-definitions.test.ts

```diff
@@ -0,0 +1,105 @@
+import serveDefinitions from "../serve-definitions.ts";
+import { Context } from "https://edge-bootstrap.netlify.app/v1/index.ts";
+import * as mf from "https://deno.land/x/mock_fetch@0.3.0/mod.ts";
+import { assertEquals } from "https://deno.land/std@0.208.0/assert/assert_equals.ts";
+
+const metricURL = "https://metric-api.eu.newrelic.com/metric/v1";
+
+const validRequests = [
+  {
+    requestURL: "https://asyncapi.com/definitions/2.4.0/info.json",
+    responseURL:
+      "https://raw.githubusercontent.com/asyncapi/spec-json-schemas/master/definitions/2.4.0/info.json",
+  },
+  {
+    requestURL: "https://asyncapi.com/schema-store/2.5.0-without-$id.json",
+    responseURL:
+      "https://raw.githubusercontent.com/asyncapi/spec-json-schemas/master/schemas/2.5.0-without-$id.json",
+  },
+];
+
+const invalidRequests = [
+  {
+    requestURL: "https://asyncapi.com/definitions/asyncapi.yaml",
+  },
+  {
+    requestURL: "https://asyncapi.com/schema-store/2.4.0.JSON",
+  },
+  {
+    requestURL: "https://asyncapi.com/foobar",
+  }
+];
+
+const context = {
+  next: () => {},
+  log: () => {},
+};
+
+let metricCalls = 0;
+
+function setup() {
+  mf.install();
+
+  mf.mock("*", (req) => {
+    console.log(req.url);
+    
+    if( req.url === metricURL ) {
+      metricCalls++;
+    }
+
+    const body = {
```

Comment:
In order to mock this as closest to the reality as possible, You need to return a Response object with a fake body representing the JSON Schema (could be just `{}`), or empty if it's to New Relic (or I believe you could just leave the same, it doesn't matter). Then set the proper Content-Type to `application/json` and proper status.

Then you can assert on the Response object directly.

Reply:
It would be awesome if we can also test the metrics, if we send the right metrics and also the 304 error NR can give https://github.com/asyncapi/website/blob/5ae285104c7829de734b96336fcd4a401dcc8c29/netlify/edge-functions/serve-definitions.ts#L52-L55

WDYT?

Reply:
> In order to mock this as closest to the reality as possible, You need to return a Response object with a fake body representing the JSON Schema (could be just `{}`), or empty if it's to New Relic (or I believe you could just leave the same, it doesn't matter). Then set the proper Content-Type to `application/json` and proper status.
> 
> Then you can assert on the Response object directly.

It can be done but what I want to do is check if the rewrite URL is correct hence the current approach.

Reply:
If you create a response object you will directly assert on the response object by equality. Is that possible?

Regarding testing metrics request (payload sent, etc), are you willing to do it?

Reply:
Let me try it and I will let you know.